descwl/analysis.py

```diff
@@ -830,47 +873,64 @@ def finalize(self,verbose,trace,calculate_bias,no_analysis):
-                ('snr_grpf',np.float32),
-                ('ds',np.float32),
-                ('dg1',np.float32),
-                ('dg2',np.float32),
-                ('ds_grp',np.float32),
-                ('dg1_grp',np.float32),
-                ('dg2_grp',np.float32),
-                ('cond_num', np.float32), #condition number of individual galaxy fisher matrix. 
-                ('cond_num_grp', np.float32), #condition number (using 2-norm) from fisher matrix of corresponding group. 
-                # HSM analysis results.
-                ('hsm_sigm',np.float32),
-                ('hsm_e1',np.float32),
-                ('hsm_e2',np.float32),
-                # Systematics fit results.
-                ('g1_fit',np.float32),
-                ('g2_fit',np.float32),
-                ]
-
-        if calculate_bias:
+                ('snr_sky',np.float32)
+            ]
+
+        #from here on, only add relevant entries. 
+        if not self.no_fisher and not self.no_analysis: 
+            dtype.extend([
+            ('snr_iso',np.float32),
+            ('snr_grp',np.float32),
+            ('snr_isof',np.float32),
+            ('snr_grpf',np.float32),
+            ('ds',np.float32),
+            ('dg1',np.float32),
+            ('dg2',np.float32),
+            ('ds_grp',np.float32),
+            ('dg1_grp',np.float32),
+            ('dg2_grp',np.float32),
+            ('cond_num', np.float32), #condition number of individual galaxy fisher matrix. 
+            ('cond_num_grp', np.float32), #condition number (using 2-norm) from fisher matrix of corresponding group.
+            ])
+
+        if self.calculate_bias and not self.no_analysis:
+            dtype.extend([
+            #fisher bias calculation results. 
+            ('bias_f', np.float32),
+            ('bias_s', np.float32),
+            ('bias_g1',np.float32),
+            ('bias_g2',np.float32),
+            ('bias_x',np.float32),
+            ('bias_y',np.float32),
+            ('bias_f_grp', np.float32),
+            ('bias_s_grp', np.float32),
+            ('bias_g1_grp',np.float32),
+            ('bias_g2_grp',np.float32),
+            ('bias_x_grp',np.float32),
+            ('bias_y_grp',np.float32)
+            ])
+
+        if not self.no_hsm and not self.no_analysis:
             dtype.extend([
-                ('bias_f', np.float32),
-                ('bias_s', np.float32),
-                ('bias_g1',np.float32),
-                ('bias_g2',np.float32),
-                ('bias_x',np.float32),
-                ('bias_y',np.float32),
-                ('bias_f_grp', np.float32),
-                ('bias_s_grp', np.float32),
-                ('bias_g1_grp',np.float32),
-                ('bias_g2_grp',np.float32),
-                ('bias_x_grp',np.float32),
-                ('bias_y_grp',np.float32),
-                ])
-
-        data = np.empty(num_galaxies, dtype=dtype)
-        if no_analysis:
-            # Return empty data table
+            # HSM analysis results.
```

Comment:
I like the change! This makes the output format vary with the options but at the same time avoids adding NaN columns that are kind of useless.

Reply:
Yeah exactly , I also thought it was unnecessary.